### PR TITLE
Early QP structure

### PR DESCRIPTION
**Pull request by tmpr**

### CODE REVIEW

File: quadratic/base.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    # Unpack problem.
+    A = problem.A
+    b = problem.b
+    c = problem.c
+    G = problem.G
+
+    # Compute H as defined on page 462. Needs to be non-singular, hence identity.
+    H = np.eye(G)
+
+    x = solve(A, b)
```
Comment (rendstrasser) [inline quadratic/base.py:54]: Requires A to be symmetric and positive-definite.
This does not need to be the case here.
I would just use the inverse-based implementation for now.

The method should be documented that it requires A to be symmetric and positive-definite, FYI @franziska-denk .

File: quadratic/base.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    Z = NotImplemented
```
Comment (rendstrasser) [inline quadratic/base.py:56]: Maybe I can help.

If we have n variables and m constraints, then we subdivide A into B and N, where B represents m (linearly indep.) components and N represents the rest of matrix A, i.e., n-m components (see 15.7). I would just select the first m columns of A as we assume A to have full-row-rank anyways. 

Then, Z is as defined in 15.15.

**Review by rendstrasser — CHANGES_REQUESTED** [review]
FYI tests are failing

File: quadratic/quadratic_problem.py [hunk 3]
```diff
@@ -3,20 +3,38 @@
...
     c: np.ndarray
     bias: float = 0
-    
+
     def __post_init__(self):
         super(QuadraticProblem, self).__post_init__()
-        self.f = lambda x: 1/2 * x @ self.G @ x + x @ self.c + self.bias
+
+        if any(ev < 0 for ev in np.linalg.eigvals(self.G)):
+            raise ValueError("Indefinite Matrix not supported.")
```
[the remark this answers is not in the document]
Comment (tmpr) [reply quadratic/quadratic_problem.py:25]: I originally thought that we don't actually care about negative matrices, but I don't know why I thought that ✨

File: quadratic/quadratic_problem.py [hunk 4]
```diff
@@ -3,20 +3,38 @@
...
+from typing import Callable, Sequence
 
 import numpy as np
 
+from shared.constraints import LinearConstraint, LinearCallable
 from shared.minimization_problem import LinearConstraintsProblem
 
 
-@dataclass
+@dataclass(repr=False)
```
[the remark this answers is not in the document]
Comment (tmpr) [reply quadratic/quadratic_problem.py:14]: There was some problem when that repr was called and it showed in the tracebacks. I disabled it for now

File: quadratic/base.py (hunk 1 above)
Comment (tmpr) [reply quadratic/base.py:54]: I'll try to do the inverse method! 👍

**Review by rendstrasser — APPROVED** [review]

**Pull request merged**